src/ape/managers/chain.py

```diff
@@ -0,0 +1,360 @@
+
+    @property
+    def _network_confirmations(self) -> int:
+        return self._provider.network.required_confirmations
+
+    def range(self, start: int = 0, stop: Optional[int] = None) -> Iterator[BlockAPI]:
+        """
+        Iterate over blocks. Works similarly to python ``range()``.
+
+        Args:
+            start (int): The first block, by number, to include in the range.
+              Defaults to 0.
+            stop (Optional[int]): The block number to stop before. Also the total
+             number of blocks to get.
+
+        Returns:
+            Iterator[:class:`~ape.api.providers.BlockAPI`]
+        """
+
+        if stop is None:
+            stop = len(self)
+
+        for i in range(start, stop):
+            yield self._get_block(i)
+
+    def poll_blocks(
+        self, start: Optional[int] = None, required_confirmations: Optional[int] = None
+    ) -> Iterator[BlockAPI]:
+        """
+        Poll new blocks. Optionally set a start block to include historical blocks.
+        **NOTE**: This is a deamon method; it does not terminate.
+
+        Args:
+            start (Optional[int]): The block number to start with. Defaults to the block
+              of the ``+1`` the current block number.
+            required_confirmations (Optional[int]): The amount of confirmations to wait
+              before yielding the block.
+
+        Returns:
+            Iterator[:class:`~ape.api.providers.BlockAPI`]
+        """
+        if required_confirmations is None:
+            required_confirmations = self._network_confirmations
+
+        # Get number of last block with the necessary amount of confirmations.
+        latest_confirmed_block_number = self.height - required_confirmations
+
+        if start is not None:
+            # Front-load historically confirmed blocks.
+            yield from self.range(start, latest_confirmed_block_number + 1)
+
+        time.sleep(self._poll_wait_interval)
+
+        while True:
+            confirmable_block_number = self.height - latest_confirmed_block_number
+            if confirmable_block_number < latest_confirmed_block_number:
+                logger.error(
+                    "Chain has reorganized since returning the last block. "
+                    "Try adjusting the required network confirmations."
+                )
+            elif confirmable_block_number > latest_confirmed_block_number:
+                yield self._get_block(confirmable_block_number)
+                latest_confirmed_block_number = confirmable_block_number
+
+            time.sleep(self._poll_wait_interval)
+
+    def _get_block(self, block_id: BlockID) -> BlockAPI:
+        return self._provider.get_block(block_id)
+
+
+class AccountHistory:
+    """
+    A container mapping account addresses to the transaction from the active session.
+    """
+
+    _map: Dict[AddressType, List[ReceiptAPI]] = {}
+
+    def __getitem__(self, address: Union[AddressAPI, AddressType, str]) -> List[ReceiptAPI]:
```

Comment:
Can this method query `ProviderAPI` (via `ExplorerAPI` to get transactions indexed by account's nonce via TBD query method) to get the set of `ReceiptAPI` from those transactions?

Reply:
Obviously, this query method would not function for chains without an Explorer (or where it is not configured) since it is a non-trivial problem to identify an account's past transactions without indexing them.

Reply:
Done!